### PR TITLE
fix: support linux builds from macOS again

### PR DESCRIPTION
**Pull request by henryiii**
One possible fix for https://github.com/pypa/cibuildwheel/issues/1962. Doesn't fix Windows, though.

### CODE REVIEW

**Comment by henryiii** [comment]
macOS doesn't have gnutar as an option in it's bsdtar.

```
tar(bsdtar): manipulate archive files
First option must be a mode specifier:
  -c Create  -r Add/Replace  -t List  -u Update  -x Extract
Common Options:
  -b #  Use # 512-byte records per I/O block
  -f <filename>  Location of archive
  -v    Verbose
  -w    Interactive
Create: tar -c [options] [<file> | <dir> | @<archive> | -C <dir> ]
  <file>, <dir>  add these items to archive
  -z, -j, -J, --lzma  Compress archive with gzip/bzip2/xz/lzma
  --format {ustar|pax|cpio|shar}  Select archive format
  --exclude <pattern>  Skip files that match pattern
  -C <dir>  Change to <dir> before processing remaining files
  @<archive>  Add entries from <archive> to output
List: tar -t [options] [<patterns>]
  <patterns>  If specified, list only entries that match
Extract: tar -x [options] [<patterns>]
  <patterns>  If specified, extract only entries that match
  -k    Keep (don't overwrite) existing files
  -m    Don't restore modification times
  -O    Write entries to stdout, don't restore to disk
  -p    Restore permissions (including ACLs, owner, file flags)
bsdtar 3.5.3 - libarchive 3.5.3 zlib/1.2.12 liblzma/5.4.3 bz2lib/1.0.8
```

**Comment by mayeut** [comment]
It's not listed in `--help`, if you do `man tar`, it redirects you to man `libarchive-formats` which lists `gnutar` c.f. https://github.com/pypa/cibuildwheel/pull/1434#issuecomment-1475278406 & https://github.com/pypa/cibuildwheel/pull/1434#issuecomment-1475292175

File: cibuildwheel/oci_container.py [hunk 1]
```diff
@@ -229,7 +229,7 @@ def copy_into(self, from_path: Path, to_path: PurePath) -> None:
         if from_path.is_dir():
             self.call(["mkdir", "-p", to_path])
             subprocess.run(
-                f"tar cf - . | {self.engine.name} exec -i {self.name} tar --no-same-owner -xC {shell_quote(to_path)} -f -",
+                f"tar -c --format ustar -f - . | {self.engine.name} exec -i {self.name} tar --format ustar --no-same-owner -xC {shell_quote(to_path)} -f -",
```
Comment (henryiii) [inline cibuildwheel/oci_container.py:232]:
```suggestion
                f"tar -c --format bsdtar -f - . | {self.engine.name} exec -i {self.name} tar --format bsdtar --no-same-owner -xC {shell_quote(to_path)} -f -",
```

File: cibuildwheel/oci_container.py [hunk 2]
```diff
@@ -229,7 +229,7 @@ def copy_into(self, from_path: Path, to_path: PurePath) -> None:
         if from_path.is_dir():
             self.call(["mkdir", "-p", to_path])
             subprocess.run(
-                f"tar cf - . | {self.engine.name} exec -i {self.name} tar --no-same-owner -xC {shell_quote(to_path)} -f -",
+                f"tar -c --format bsdtar -f - . | {self.engine.name} exec -i {self.name} tar --format bsdtar --no-same-owner -xC {shell_quote(to_path)} -f -",
```
Comment (henryiii) [inline cibuildwheel/oci_container.py:232]:
```suggestion
                f"tar -c --format gnutar -f - . | {self.engine.name} exec -i {self.name} tar --format gnutar --no-same-owner -xC {shell_quote(to_path)} -f -",
```

**Comment by henryiii** [comment]
```
Copying project into container...

    + mkdir -p /project
tar: gnutar: Invalid archive format
Try `tar --help' or `tar --usage' for more information.
tar: Write error

                                                              ✕ 0.78s
```

**Comment by mayeut** [comment]
It shall only be on host side of tar create for macOS only (I pushed a commit to that effect).

**Pull request closed without merging**